lib/WTSI/NPG/iRODS/DataObject.pm

```diff
@@ -8,23 +8,37 @@ use Set::Scalar;
 use Try::Tiny;
 
 use WTSI::NPG::iRODS;
+use WTSI::NPG::iRODS::Metadata qw($SAMPLE_CONSENT
+                                  $SAMPLE_CONSENT_WITHDRAWN);
+use WTSI::NPG::iRODS::Replicate;
+use WTSI::NPG::iRODS::Types qw(ArrayRefOfReplicate);
 
 our $VERSION = '';
 
 with 'WTSI::NPG::iRODS::Path';
 
 has 'data_object' =>
-  (is        => 'ro',
-   isa       => 'Str',
-   required  => 1,
-   lazy      => 1,
-   default   => q{.},
-   predicate => 'has_data_object');
-
-has 'checksum' => (is        => 'rw',
-                   isa       => 'Str',
-                   predicate => 'has_checksum',
-                   clearer   => 'clear_checksum');
+  (is            => 'ro',
+   isa           => 'Str',
+   required      => 1,
+   lazy          => 1,
+   default       => q{.},
+   predicate     => 'has_data_object',
+   documentation => 'The data object component of the iRODS path.');
+
+has 'checksum' =>
+  (is            => 'rw',
+   isa           => 'Str',
+   predicate     => 'has_checksum',
+   clearer       => 'clear_checksum',
+   documentation => 'The checksum of thie data object.');
+
+has 'replicates' =>
+  (is            => 'rw',
```

Comment:
```rw``` for attributes ```replicates``` and ```checksum``` together with an ```around``` to obtain data looks perfectly functional and fits in - but how about ```ro```, ```lazy``` and a builder? Better/same/worse?